<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SelectOutput</class>
 <widget class="QDialog" name="SelectOutput">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>264</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Select Output Device</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QRadioButton" name="midiDriverButton">
        <property name="text">
         <string>Windows MIDI Driver</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="midiPortSelect">
        <item>
         <property name="text">
          <string>Select MIDI Output Port</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QRadioButton" name="midiFileButton">
        <property name="text">
         <string>Export to MIDI Type 0 File</string>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSplitter" name="splitter">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <widget class="QLineEdit" name="fileSelect">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="placeholderText">
          <string>Output File Path</string>
         </property>
        </widget>
        <widget class="QPushButton" name="browseButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="maximumSize">
          <size>
           <width>65</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>Browse...</string>
         </property>
        </widget>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QRadioButton" name="USBButton">
        <property name="text">
         <string>Output to USB Device</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="comPortSelect">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <item>
         <property name="text">
          <string>Select Serial Output Port</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SelectOutput</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SelectOutput</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>midiFileButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>fileSelect</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>40</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>64</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>USBButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>comPortSelect</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>181</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>205</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>midiDriverButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>midiPortSelect</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>40</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>64</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>midiFileButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>browseButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>348</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>browseButton</sender>
   <signal>clicked()</signal>
   <receiver>SelectOutput</receiver>
   <slot>browse()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>348</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>midiDriverButton</sender>
   <signal>clicked()</signal>
   <receiver>SelectOutput</receiver>
   <slot>checkValid()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>40</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>midiFileButton</sender>
   <signal>clicked()</signal>
   <receiver>SelectOutput</receiver>
   <slot>checkValid()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>USBButton</sender>
   <signal>clicked()</signal>
   <receiver>SelectOutput</receiver>
   <slot>checkValid()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>181</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileSelect</sender>
   <signal>textChanged(QString)</signal>
   <receiver>SelectOutput</receiver>
   <slot>checkValid()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>164</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comPortSelect</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>SelectOutput</receiver>
   <slot>checkValid()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>205</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>midiPortSelect</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>SelectOutput</receiver>
   <slot>checkValid()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>64</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>browse()</slot>
  <slot>checkValid()</slot>
 </slots>
</ui>
